File: DataStructures/DoubleLinkedList/Node.cpp
#include "Node.h"
Node::Node(int val,Node* parent) {
	this->value = val;
	this->parent = parent;
	this->child = nullptr;
}
bool Node::Contains(int val) {
	if (this->value == val) return true;
	if(this->child != nullptr){
		return this->child->Contains(val);
	}
	return false;
}
void Node::Remove(int val) {
	if (this->child == nullptr) return;
	if (this->child->value == val) {
		Node* nodeToDelete = this->child;
		this->child = this->child->child;
		delete nodeToDelete;
		nodeToDelete = nullptr;
	}
	else {
		this->child->Remove(val);
	}
}
void Node::CleanUp() {
	if (this->child != nullptr) {
		this->child->CleanUp();
	}
	delete this->child;
	this->child = nullptr;
	this->parent = nullptr;
}

File: src/datastructures/linkedlist/double/Node.php
<?php
declare(strict_types=1);
namespace ZimCodes\Self_Learning\DataStructures\LinkedList\Double;
use JetBrains\PhpStorm\Pure;
use ZimCodes\Self_Learning\DataStructures\LinkedList;
use ZimCodes\Self_Learning\DataStructures\LinkedList\INode;

class Node extends LinkedList\Node{
    #[Pure] public function __construct(mixed $value, private ?INode $parent = null, ?INode $child = null){
        parent::__construct($value,$child);
    }
    public function delete(mixed $value): void{
        if($this->child === null) return;
        if($value !== $this->child->getValue()){
            $this->child->delete($value);
        }else{
            $grandChild = $this->child->getChild();
            $this->child = $grandChild;
            $grandChild->setParent($this);
        }
    }
    public function getParent():?INode{
        return $this->parent;
    }
    public function setParent(?INode $parent):void{
        $this->parent = $parent;
    }
}